src/Auth/BaseAuthenticate.php

```diff
@@ -146,6 +146,9 @@ protected function _query($username)
         if (!empty($config['contain'])) {
             $options['contain'] = $config['contain'];
         }
+        if (empty($config['username'])) {
```

Comment:
@Phillaf In that case you need to check `$options['username']` after the `if` block for `is_array($finder)` below.